### PR TITLE
Cleanup dependencies

### LINKED ISSUE #794: unpin pandas<1.3
* xclim version: 0.28.1
* Python version: 3.9.6
* Operating System: Mac

### Description

I bumped my conda environment today and I realised my pandas version got downgraded. From running `mamba repoquery whoneeds pandas` I noticed it's pinned to <1.3
https://github.com/Ouranosinc/xclim/blob/master/setup.py#L26
in this repo.

I had a quick look in the issue tracker and couldn't find any issues on this so wanted to raise this here. There may be good reason why it's pinned. Happy to help work on unpinning it if it's desired.

### LINKED ISSUE #799: Notebooks are not rendering in the docs
See https://xclim.readthedocs.io/en/latest/notebooks/index.html

This also happens locally. Seems to be related with the latest sphinx relase (4.1.2, Jul 26), but I'm not sure how.

### PR DESCRIPTION
**Pull request by aulemahal**
### Pull Request Checklist:
- [x] This PR addresses an already opened issue (for bug fixes / features)
    - This PR fixes #799, fixes #794 and addresses discussions from the internal slack.
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Documentation has been added / updated (for bug fixes / features)
- [x] HISTORY.rst has been updated (with summary of main changes)
- [ ] `bumpversion (major / minor / patch)` has been called on this branch
- [ ] Tags have been pushed (`git push --tags`)
- [ ] The relevant author information has been added to `.zenodo.json`

### What kind of change does this PR introduce?

* Removes all the "extra" dependencies from `setup.py` except "dev". All tools needed for testing, documenting and developping xclim are listed in `requirements_dev.py`.
* Fixes a doctest issue triggered by cftime 1.5.
* Removes the line that was installing cftime 1.4.1 in the test suite (!!!!)
* Unpin pandas
* Removes the mention of `".ipynb"` as a source suffix in `docs/conf.py`. That variable, when it is a list, will associate all those files with the RST parser. Before `nbsphinx` 0.8.7 (10 Aug 2021), the plugin would override the current configuration, but now it doesn't, so the bug showed itself.

### Does this PR introduce a breaking change?
I don't know if this count as a "breaking change", but installation options have changed.

### Other information:

@Zeitsperre I removed some packages from the dev reqs:
```
ipykernel, ipython, jupyter_client, wheel, watchdog, twine
```
I was able to follow all the steps of the "contributing" page of the doc. Are any of those explicitely needed by developers nonetheless? I see that some are needed for packaging, but I think we can leave them out of the dev reqs, as only you or few others will ever do that task.

I haven't addressed #795, but could do it here if people agree with the idea.

### CODE REVIEW

File: tox.ini
Comment (Zeitsperre) [inline tox.ini:64]: I would re-add coveralls. It would be good to get this check working again.

Comment (aulemahal) [reply tox.ini:64]: haha so polite. My bad!

File: docs/conf.py
Comment (Zeitsperre) [inline docs/conf.py:132]: Set it and forget it?
```suggestion
copyright = f"2018-{datetime.datetime.now().year}, Ouranos Inc., Travis Logan, and contributors"
```

File: HISTORY.rst
Comment (Zeitsperre) [inline HISTORY.rst:22]:
```suggestion
* Removed the `"gis", "docs", "test" and "setup"`extra dependencies from ``setup.py``. The ``dev`` recipe now includes all tools needed for xclim's development.
```

**Review by Zeitsperre — APPROVED** [review]
This is looking much nicer. Nice!

**Comment by Zeitsperre** [comment]
> @Zeitsperre I removed some packages from the dev reqs:
> 
> ```
> ipykernel, ipython, jupyter_client, wheel, watchdog, twine
> ```
> 
> I was able to follow all the steps of the "contributing" page of the doc. Are any of those explicitely needed by developers nonetheless? I see that some are needed for packaging, but I think we can leave them out of the dev reqs, as only you or few others will ever do that task.

`wheel` and `twine` are only needed for uploading releases. Seeing as this is a major library of ours, it would be good to have the steps in the development docs. Feel free to keep them removed for now, but I'll open an issue to finish the packaging docs on another PR.

For the Jupyter-related libraries, we need a minimal environment in the `dev` recipe for running the notebooks via pytest at the very least. If we can get away without those libraries, great!

I had no idea watchdog was in there. Is that library used for some form of provenance tracking?

> I haven't addressed #795, but could do it here if people agree with the idea.

I have some thoughts on this, but I'll add to the issue. My vote is to hold off for now.

File: docs/conf.py
Comment (Zeitsperre) [reply docs/conf.py]: Woops! This is going to need an import of `datetime`! My bad!

**Comment by Zeitsperre** [comment]
After our discussion from the other day, we'll go forward with the suggestion of bundling the testing dependencies into the GitHub source `environment.yml`. Once that's in, we should be good to merge!

**Pull request merged**
